lib/matplotlib/backend_bases.py

```diff
@@ -400,7 +400,7 @@ def _iter_collection_uses_per_path(self, paths, all_transforms,
         if Npaths == 0 or (len(facecolors) == 0 and len(edgecolors) == 0):
             return 0
         Npath_ids = max(Npaths, len(all_transforms))
-        N = max(Npath_ids, len(offsets))
+        N = max(Npaths, len(offsets))
```

Comment:
If I am not mistaken, this function is only used to compute whether or not to do an optimization. Hence, over estimating the number of paths drawn is not that important and I am assuming that this as no impact on the bug you are fixing?